### PR TITLE
[AIRFLOW-5814] Implementing Presto hook tests

### PR DESCRIPTION
**Pull request by blcksrx**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Implementing presto_hook tests methods including tests for **get_first_record**, **get_records** and **get_pandas_df**

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by OmerJog** [comment]
can you also add tests for the connection?
Similar to what we have for other dbs
https://github.com/apache/airflow/blob/master/tests/hooks/test_mysql_hook.py#L37
https://github.com/apache/airflow/blob/master/tests/hooks/test_postgres_hook.py#L29

**Comment by blcksrx** [comment]
> can you also add tests for the connection?
> Similar to what we have for other dbs
> https://github.com/apache/airflow/blob/master/tests/hooks/test_mysql_hook.py#L37
> https://github.com/apache/airflow/blob/master/tests/hooks/test_postgres_hook.py#L29

I will create separates PR's for them, is it okay?

**Comment by OmerJog** [comment]
@bslakter that's ur call I think but I think it makes sense to add it in the same PR.

**Comment by blcksrx** [comment]
> @bslakter that's ur call I think but I think it makes sense to add it in the same PR.

Okay! I will rename my PR and also the Jira issue and  then i will add them to this PR. Is it okay?

File: tests/hooks/test_presto_hook.py [hunk 1]
```diff
@@ -51,3 +51,39 @@ def test_insert_rows(self, mock_insert_rows):
...
+    def test_get_records(self):
+        statement = 'SQL'
+        result_sets = [('row1',), ('row2',)]
+        self.cur.fetchall.return_value = result_sets
+
+        self.assertEqual(result_sets, self.db_hook.get_records(statement))
+        assert self.conn.close.call_count == 1
+        assert self.cur.close.call_count == 1
+        self.cur.execute.assert_called_once_with(statement)
+        print(1)
```
Comment (feluelle) [inline tests/hooks/test_presto_hook.py]: Please remove this print statement ;)

File: tests/hooks/test_presto_hook.py [hunk 2]
```diff
@@ -51,3 +51,39 @@ def test_insert_rows(self, mock_insert_rows):
...
         self.db_hook.insert_rows(table, rows, target_fields)
         mock_insert_rows.assert_called_once_with(table, rows, None, 0)
+
+    def test_get_first_record(self):
+        statement = 'SQL'
+        result_sets = [('row1',), ('row2',)]
+        self.cur.fetchone.return_value = result_sets[0]
+
+        self.assertEqual(result_sets[0], self.db_hook.get_first(statement))
+        assert self.conn.close.call_count == 1
```
Comment (feluelle) [inline tests/hooks/test_presto_hook.py]:
```suggestion
        self.conn.close.assert_called_once_with()
```

File: tests/hooks/test_presto_hook.py [hunk 3]
```diff
@@ -51,3 +51,39 @@ def test_insert_rows(self, mock_insert_rows):
...
+        assert self.cur.close.call_count == 1
```
Comment (feluelle) [inline tests/hooks/test_presto_hook.py]:
```suggestion
        self.cur.close.assert_called_once_with()
```

File: tests/hooks/test_presto_hook.py [hunk 4]
```diff
@@ -51,3 +51,39 @@ def test_insert_rows(self, mock_insert_rows):
...
+        self.cur.execute.assert_called_once_with(statement)
+
+    def test_get_records(self):
+        statement = 'SQL'
+        result_sets = [('row1',), ('row2',)]
+        self.cur.fetchall.return_value = result_sets
+
+        self.assertEqual(result_sets, self.db_hook.get_records(statement))
+        assert self.conn.close.call_count == 1
```
Comment (feluelle) [inline tests/hooks/test_presto_hook.py]: Same here.

File: tests/hooks/test_presto_hook.py [hunk 5]
```diff
@@ -51,3 +51,39 @@ def test_insert_rows(self, mock_insert_rows):
...
+        assert self.cur.close.call_count == 1
```
Comment (feluelle) [inline tests/hooks/test_presto_hook.py]: And here.

**Review by feluelle — CHANGES_REQUESTED** [review]
However you like @blcksrx. But like @OmerJog said it makes probably more sense to add the test for [get_conn](https://codecov.io/gh/apache/airflow/src/bd3aeb20a1640849578b6526d1c9736de2b52fdc/airflow/hooks/presto_hook.py#L44) to the same PR since you named it "Implementing Presto hook tests" you don't need to rename it.

File: tests/hooks/test_presto_hook.py (hunk 1 above)
Comment (blcksrx) [reply tests/hooks/test_presto_hook.py]: okay :)))

**Comment by blcksrx** [comment]
Actually, because of this implementation, I can't test **get_connection**. now I have two solutions for it:
1.  change the **setUp** method and also change other methods related to changing
2. do exactly the same as **test_postgres_hook** and create another class for test connections.

@feluelle 
```
def setUp(self):
        super().setUp()

        self.cur = mock.MagicMock()
        self.conn = mock.MagicMock()
        self.conn.cursor.return_value = self.cur
        conn = self.conn

        class UnitTestPrestoHook(PrestoHook):
            conn_name_attr = 'test_conn_id'

            def get_conn(self):
                return conn

        self.db_hook = UnitTestPrestoHook()
```

File: airflow/www/security.py [hunk 6]
```diff
@@ -21,7 +21,7 @@
 from flask import g
 from flask_appbuilder.security.sqla import models as sqla_models
 from flask_appbuilder.security.sqla.manager import SecurityManager
-from sqlalchemy import or_, and_
+from sqlalchemy import and_, or_
```
Comment (OmerJog) [inline airflow/www/security.py:24]: Looks like unrelated change?

Comment (blcksrx) [reply airflow/www/security.py:24]: That was related to an issue that were isor error on ci/cd and i  did rebase the git and force push it

Comment (blcksrx) [reply airflow/www/security.py:24]: I will close this PR and create another on to fix this.
